src/relations/db.py

```diff
@@ -91,19 +93,19 @@ def _on_relation_changed(self, event: RelationChangedEvent) -> None:
         application_relation_databag = event.relation.data[self.charm.app]
 
         # Do not allow apps requesting extensions to be installed.
-        if "extensions" in event.relation.data[
-            event.app
-        ] or "extensions" in event.relation.data.get(event.unit, {}):
+        if "extensions" in event.relation.data.get(
+            event.app, {}
+        ) or "extensions" in event.relation.data.get(event.unit, {}):
```

Comment:
Might as well go all the way and assume either `event.app` or `event.unit` can be None